### PR TITLE
[CELEBORN-977] Support RocksDB as recover DB backend

### PR DESCRIPTION
**Pull request by cxzl25**
### Why are the changes needed?

LevelDB does not support mac arm version.

```java
java.lang.UnsatisfiedLinkError: Could not load library. Reasons: [no leveldbjni64-1.8 in java.library.path, no leveldbjni-1.8 in java.library.path, no leveldbjni in java.library.path, /private/var/folders/tc/r2n_8g6j4731h7clfqwntg880000gn/T/libleveldbjni-64-1-4616234670453989010.8: dlopen(/private/var/folders/tc/r2n_8g6j4731h7clfqwntg880000gn/T/libleveldbjni-64-1-4616234670453989010.8, 0x0001): tried: '/private/var/folders/tc/r2n_8g6j4731h7clfqwntg880000gn/T/libleveldbjni-64-1-4616234670453989010.8' (fat file, but missing compatible architecture (have 'x86_64,i386', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/tc/r2n_8g6j4731h7clfqwntg880000gn/T/libleveldbjni-64-1-4616234670453989010.8' (no such file), '/private/var/folders/tc/r2n_8g6j4731h7clfqwntg880000gn/T/libleveldbjni-64-1-4616234670453989010.8' (fat file, but missing compatible architecture (have 'x86_64,i386', need 'arm64'))]
  	at org.fusesource.hawtjni.runtime.Library.doLoad(Library.java:182)
  	at org.fusesource.hawtjni.runtime.Library.load(Library.java:140)
  	at org.fusesource.leveldbjni.JniDBFactory.<clinit>(JniDBFactory.java:48)
  	at org.apache.celeborn.service.deploy.worker.shuffledb.LevelDBProvider.initLevelDB(LevelDBProvider.java:49)
  	at org.apache.celeborn.service.deploy.worker.shuffledb.DBProvider.initDB(DBProvider.java:30)
  	at org.apache.celeborn.service.deploy.worker.storage.StorageManager.<init>(StorageManager.scala:197)
  	at org.apache.celeborn.service.deploy.worker.Worker.<init>(Worker.scala:109)
  	at org.apache.celeborn.service.deploy.worker.Worker$.main(Worker.scala:734)
  	at org.apache.celeborn.service.deploy.worker.Worker.main(Worker.scala)
```

The released `leveldbjni-all` for `org.fusesource.leveldbjni` does not support AArch64 Linux, we need to use `org.openlabtesting.leveldbjni`.

See https://issues.apache.org/jira/browse/HADOOP-16614

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
local test

### CODE REVIEW

**Comment by pan3793** [comment]
LICENSE related stuff need to update

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2593,12 +2595,20 @@ object CelebornConf extends Logging {
...
       .transform(_.replace("<tmp>", System.getProperty("java.io.tmpdir")))
       .createWithDefault(s"<tmp>/recover")
 
+  val WORKER_GRACEFUL_SHUTDOWN_RECOVER_DB_BACKEND: ConfigEntry[String] =
+    buildConf("celeborn.worker.graceful.shutdown.recoverDbBackend")
+      .categories("worker")
+      .doc("Specifies a disk-based store used in local db. LEVELDB or ROCKSDB.")
+      .version("0.4.0")
+      .stringConf
+      .createWithDefault("LEVELDB")
```
Comment (jiaoqingbo) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2610]: checkValues(Set("LEVELDB", "ROCKSDB"))？

File: LICENSE [hunk 2]
```diff
@@ -211,3 +211,4 @@ Apache Software Foundation License 2.0
 --------------------------------------
 
 common/src/main/java/org/apache/celeborn/common/network/util/LimitedInputStream.java
+worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/*
```
Comment (pan3793) [inline LICENSE:214]: we don't need such an exclusion for copying code from apache projects.

**Review by pan3793 — APPROVED** [review]

**Review by jiaoqingbo — APPROVED** [review]

File: project/CelebornBuild.scala [hunk 3]
```diff
@@ -403,6 +405,7 @@ object CelebornWorker {
         Dependencies.log4j12Api,
         Dependencies.log4jSlf4jImpl,
         Dependencies.leveldbJniAll,
+        Dependencies.rocksdbJni,
```
Comment (cfmcgrady) [inline project/CelebornBuild.scala:408]: nit: sort alphabetically

**Review by cfmcgrady — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to main

**Comment by waitinfuture** [comment]
LGTM, thanks!

**Pull request closed without merging**